voorhoede/head-start
# refactor: middleware as atomic files + docs

l# Changes

Refactors middleware to atomic files (`src/middleware/...`), so middleware stays maintainable when adding more (like the upcoming file proxy middleware) and has a structure suitable to add tests to (in `src/middleware/tests/` like done in `src/lib/datocms/tests/`).

# Associated issue

N/A

# How to test

1. Open preview link
2. Verify the different middleware handlers (`i18n`, `redirects`) still work (`preview` can't be tested on deploy preview, unless we list it as a preview branch; `datocms` is hard to check on deploy preview)
3. Run locally
4. Verify the different middleware handlers (`datocms`, `i18n`, `preview`, `redirects`) still work

# Checklist

- ~I have added a decision log entry if the change affects the architecture or changes a significant technology~
